Product creation using package photo: OCR library integration

0nko opened the pull request:
This PR integrates the ML Kit library for text recognition, which is used to display keywords for product title and description generation (next PR). Also, the loading and error state UI will be added in the next PR.

This PR also updates the README to reflect that we are using ML Kit library.

**To test:**
1. Go to Products
2. Tap on + FAB
3. Select the AI option
4. Tap on Use package photo button
5. Use the media library to select a picture of a product with some text
6. Notice the image is loaded and a list of keywords from the image is displayed
7. Try using a different image

<img src="https://github.com/woocommerce/woocommerce-android/assets/1522856/433d9a86-cf5e-4c2f-a763-856ac97e224c" width="500" />

*Note:* Please merge #10003 first.

0nko committed:
Handle the media library image source

0nko committed:
Fix the package photo bottom sheet UI layout

0nko committed:
Add loading and animation to package image

0nko committed:
Fix ktlint issues

0nko committed:
Add dependencies

0nko committed:
Fix the package photo image scaling

0nko committed:
Add ML Kit text recognition for keyword extraction

0nko committed:
Address ktlint issue

0nko committed:
Merge branch 'feature/media-picker-library' into feature/package-photo-ocr

# Conflicts:
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/PackagePhotoBottomSheet.kt
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/PackagePhotoViewModel.kt

0nko committed:
Update the dependency version with a common variable

0nko commented on WooCommerce/build.gradle:
```diff
@@ -348,6 +349,7 @@ dependencies {
     // Coroutines
     implementation "org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutinesVersion"
     implementation "org.jetbrains.kotlinx:kotlinx-coroutines-android:$coroutinesVersion"
+    implementation "org.jetbrains.kotlinx:kotlinx-coroutines-play-services:$coroutinesVersion"
```
This dependency adds an extension function that can convert the `Task` object of the Play Services request to a coroutine.

0nko committed:
Update the README file to include the mention of ML Kit

0nko committed:
Merge branch 'feature/media-picker-library' into feature/package-photo-ocr

0nko committed:
Make the preview method private

JorgeMucientes commented:
Great work @0nko. The feature works pretty well! I just notice a minor UI glitch for text in darkmode: 

https://github.com/woocommerce/woocommerce-android/assets/2663464/874951dc-31a7-4038-9ae6-23399ce7d1b2

Once that is addressed feel free to ship :shipit:

0nko committed:
Merge branch 'feature/media-picker-library' into feature/package-photo-ocr

0nko committed:
Fix the checkbox color for night mode

0nko committed:
Fix the keyword item text color in night mode

0nko commented:
Thank you @JorgeMucientes. I've addressed your comment.

<img src="https://github.com/woocommerce/woocommerce-android/assets/1522856/fa6695e5-5949-4c23-a2bc-afea5e3e7b7b" width="350" />
